internal/onboarding/ms365.go

```diff
@@ -0,0 +1,221 @@
+	if integration.Name == "" {
+		integration.Name = generateAzureIntegrationName(integration.Primary)
+		log.Info().Msgf(
+			"integration name not provided, using %s",
+			theme.DefaultTheme.Primary(integration.Name),
+		)
+	}
+
+	// Generate required providers block
+	requiredProvidersBlock, err := tfgen.CreateRequiredProviders(
+		tfgen.NewRequiredProvider("mondoo",
+			tfgen.HclRequiredProviderWithSource(mondooProviderSource),
+			tfgen.HclRequiredProviderWithVersion(mondooProviderVersion),
+		),
+	)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate required providers")
+	}
+
+	featuresBlock, err := tfgen.HclCreateGenericBlock("features", []string{}, nil)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to append features block to azurerm provider")
+	}
+	selfSignedCertSubjectBlock, err := tfgen.HclCreateGenericBlock("subject", nil,
+		tfgen.Attributes{"common_name": "mondoo"},
+	)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate self signed cert subject block")
+	}
+	requiredResourceAccessBlock1, err := createResourceAccessBlock(
+		"00000003-0000-0000-c000-000000000000",
+		[]map[string]interface{}{
+			{"id": "246dd0d5-5bd0-4def-940b-0421030a5b68", "type": "Role"},
+			{"id": "e321f0bb-e7f7-481e-bb28-e3b0b32d4bd0", "type": "Role"},
+			{"id": "5e0edab9-c148-49d0-b423-ac253e121825", "type": "Role"},
+			{"id": "bf394140-e372-4bf9-a898-299cfc7564e5", "type": "Role"},
+			{"id": "dc377aa6-52d8-4e23-b271-2a7ae04cedf3", "type": "Role"},
+			{"id": "9e640839-a198-48fb-8b9a-013fd6f6cbcd", "type": "Role"},
+			{"id": "37730810-e9ba-4e46-b07e-8ca78d182097", "type": "Role"},
+		},
+	)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate first resource access block")
+	}
+
+	requiredResourceAccessBlock2, err := createResourceAccessBlock(
+		"00000003-0000-0ff1-ce00-000000000000",
+		[]map[string]interface{}{
+			{"id": "678536fe-1083-478a-9c59-b99265e6b0d3", "type": "Role"},
+		},
+	)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate second resource access block")
+	}
+
+	requiredResourceAccessBlock3, err := createResourceAccessBlock(
+		"00000002-0000-0ff1-ce00-000000000000",
+		[]map[string]interface{}{
+			{"id": "dc50a0fb-09a3-484d-be87-e023b12c6440", "type": "Role"},
+		},
+	)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate third resource access block")
+	}
+
+	var (
+		providerAzureAD = tfgen.NewProvider("azuread")
+		providerAzureRM = tfgen.NewProvider("azurerm",
+			tfgen.HclProviderWithAttributes(tfgen.Attributes{"subscription_id": integration.Primary}),
+			tfgen.HclProviderWithGenericBlocks(featuresBlock),
+		)
+		providerMondoo = tfgen.NewProvider("mondoo", tfgen.HclProviderWithAttributes(
+			tfgen.Attributes{"space": integration.Space},
+		))
+		dataADClientConfig    = tfgen.NewDataSource("azuread_client_config", "current")
+		resourceAdApplication = tfgen.NewResource("azuread_application", "mondoo",
```

Comment:
We might not need to create an AD application, if you look at how we did it for Azure, we used the `azuread_directory_role` and `azuread_directory_role_assignment` resources. Maybe we can do the same for this integration. 

I am going to spend some time testing and giving you an example but, if you have cycles, please give it a try. I quick google search gave back this in addition to the `Directory Readers` you will find in azure:
```
resource "azuread_directory_role" "exchange_admin" {
  display_name = "Exchange Administrator"
}

resource "azuread_directory_role_assignment" "user_exchange_admin" {
  principal_object_id = "<user_object_id>"  # Replace with the user's object ID
  role_id             = azuread_directory_role.exchange_admin.object_id
}
```